### PR TITLE
feat: Add template pull cmd

### PR DESCRIPTION
**Pull request by sreya**
This PR adds a `coder templates pull` command to inspect a template. Some behavior notes:

- It only returns the latest version of a template
- Not specifying a destination will cause the command to write to `stdout` similar to any linux utility
- It does not attempt to untar the archive.

### CODE REVIEW

File: cli/clitest/clitest.go [hunk 1]
```diff
@@ -46,6 +48,29 @@ func CreateTemplateVersionSource(t *testing.T, responses *echo.Responses) string
...
+
+			{
+				Type: &proto.Parse_Response_Complete{
+					Complete: &proto.Parse_Complete{},
+				},
+			},
+		},
+		Provision: echo.ProvisionComplete,
+	}
+}
```
Comment (kylecarbs) [inline cli/clitest/clitest.go:72]: This isn't _exactly_ generalizable, so I might keep this in your test.

This is typically used to pass agent tokens, create fake resources, apps, etc.

File: cli/templatepull.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/cliui"
+	"github.com/coder/coder/codersdk"
+)
+
+func templatePull() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "pull <name> [destination]",
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply cli/templatepull.go:18]: It can default to `latest` tho

File: coderd/provisionerjobs.go [hunk 3]
```diff
@@ -287,9 +287,10 @@ func convertProvisionerJobLog(provisionerJobLog database.ProvisionerJobLog) code
 
 func convertProvisionerJob(provisionerJob database.ProvisionerJob) codersdk.ProvisionerJob {
 	job := codersdk.ProvisionerJob{
-		ID:        provisionerJob.ID,
-		CreatedAt: provisionerJob.CreatedAt,
-		Error:     provisionerJob.Error.String,
+		ID:         provisionerJob.ID,
+		CreatedAt:  provisionerJob.CreatedAt,
+		Error:      provisionerJob.Error.String,
+		SourceHash: provisionerJob.StorageSource,
```
Comment (kylecarbs) [inline coderd/provisionerjobs.go:293]: I think we keep this as `StorageSource` for rn for API <-> DB naming consistency

**Review by kylecarbs — APPROVED** [review]

File: cli/templatepull.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sreya) [reply cli/templatepull.go:18]: I figured I'd leave that out until we have a good way of referencing template versions. Right now they have "names" but they appear to be optional and I want to change it to just be an auto-incrementing number.

[the remark this answers is not in the document]
Comment (kylecarbs) [reply cli/templatepull.go:18]: Reasonable

**Pull request merged**